lib/instrumentation/http-shared.js

```diff
@@ -182,33 +182,35 @@ exports.traceOutgoingRequest = function (agent, moduleName, method) {
         return emit.apply(req, arguments)
       }
 
-      let url
+      let url, statusCode
       httpRequestToUrl(req).then(_url => {
         url = _url
       }).catch(() => {
         agent.logger.warn('unable to identify http.ClientRequest url %o', { id: id })
       })
 
-      return req
-
-      function onresponse (res) {
-        // Work around async_hooks bug in Node.js 12.0 - 12.2 (https://github.com/nodejs/node/pull/27477)
-        ins._recoverTransaction(span.transaction)
-
-        agent.logger.debug('intercepted http.ClientRequest response event %o', { id: id })
-        ins.bindEmitter(res)
-
-        res.prependListener('end', onEnd)
+      // In case the request is ended prematurely
+      endOfStream(req, function () {
+        // End span only when request is aborted as EOS fires for
+        // 'abort', 'close', 'end' events which might end span prematurely
+        if (req.aborted) {
+          if (span.ended) return
+          agent.logger.debug('unexpected end of outgoing http request %o', { id: id })
```

Comment:
why is this unexpected? isn't it expected that the request stream is ended when the request is aborted...?

Reply:
My guess its not common to abort requests in the middle of the stream and could also be due to socket reset by peer. I will rename it like this `intercepted http.IncomingMessage abort event` to keep it in sync.